### PR TITLE
Added dialyzer locally (not on CI)

### PR DESCRIPTION
**Pull request by begedin**
# What's in this PR?

Work in progress, still trying to get it work on CI. I will also have to actually fix reported issues once it does work.

Update: Got the build to fail if there are dialyzer warnings. It also made the build quite long, so need to figure out how to add caching or something.

### CODE REVIEW

**Comment by joshsmith** [comment]
In the circle_pre_build you may need to `mix dialyzer.plt`, not positive.

**Comment by joshsmith** [comment]
The exit code `2` means:

> No problems were found during the analysis, but warnings were emitted.

**Comment by begedin** [comment]
I'm OK with the build failing on exit code 2, though, since it does mean that our specs aren't really correct.

**Comment by joshsmith** [comment]
I might suggest running the dialyxir test on a separate line. Should make Circle reporting more transparent.

File: lib/code_corps/stripe_service/stripe_connect_subscription_service.ex [hunk 1]
```diff
@@ -29,6 +34,13 @@ defmodule CodeCorps.StripeService.StripeConnectSubscriptionService do
   - If the subscription is created or found, associated project totals will get updated
   - If the subscription is created or found, associated donation goal states will be updated
   """
+  @spec find_or_create(map) :: {:ok, StripeConnectSubscription.t} |
+                               {:error, Ecto.Changeset.t} |
+                               {:error, Stripe.APIErrorResponse.t} |
+                               {:error, :project_not_found} |
+                               {:error, :project_not_ready} |
+                               {:error, :user_not_found} |
+                               {:error, :user_not_ready}
```
Comment (joshsmith) [inline lib/code_corps/stripe_service/stripe_connect_subscription_service.ex:33]: Maybe remove the doc for the return values since `@spec` does that?

**Review by joshsmith — APPROVED** [review]

File: lib/code_corps/stripe_service/stripe_connect_subscription_service.ex (hunk 1 above)
Comment (begedin) [reply lib/code_corps/stripe_service/stripe_connect_subscription_service.ex:33]: I thought I did, definitely did in the other place I added it. Done now.

**Pull request merged**